### PR TITLE
feat: 💅 Review Page improvements

### PR DESCRIPTION
**Pull request by fpigeonjr**
## What changed

- Review Page improvements
- refactor react logic into hooks module
- text updates for various accordions
- adds Review Changes Accordion and StatusChangeReviewCards
- adds Notes Accordion
- adds Services Components grouping to ReviewBLITables
- adds placeholder Documents
- various UX review items
- adds Cancel button and functionality
- Agreement form validations

## Issue

#2488 #2487

## How to test

_Write out steps for how someone could test this PR against the acceptance criteria_

## Screenshots

https://github.com/user-attachments/assets/ee23f004-715f-490c-a15a-4f717d120636

## Definition of Done Checklist
- [X] OESA: Code refactored for clarity
- [X] OESA: Dependency rules followed
- [X] Automated unit tests updated and passed
- [ ] Automated integration tests updated and passed
- [X] Automated quality tests updated and passed
- [X] Automated load tests updated and passed
- [X] Automated a11y tests updated and passed
- [X] Automated security tests updated and passed
- [X] 90%+ Code coverage achieved
- [X] Form validations updated

### CODE REVIEW

File: frontend/src/components/Agreements/AgreementChangesAccordion/AgreementChangesAccordion.jsx [hunk 1]
```diff
@@ -7,6 +7,8 @@ import { getDecimalScale } from "../../../helpers/currencyFormat.helpers";
 import SummaryCard from "../../UI/SummaryCard";
 import styles from "./small-summary-card.module.css";
 
+// NOTE: This component is not currently being used in the application.
+
```
Comment (fpigeonjr) [inline frontend/src/components/Agreements/AgreementChangesAccordion/AgreementChangesAccordion.jsx:11]: 🤔 should we delete unused components?

File: frontend/src/components/Agreements/AgreementMetaAccordion/AgreementMetaAccordion.jsx [hunk 2]
```diff
@@ -57,6 +57,19 @@ const AgreementMetaAccordion = ({ agreement, projectOfficerName, res, cn, conver
                                 className={cn("type")}
                                 value={convertCodeForDisplay("agreementType", agreement?.agreement_type)}
                             />
+                            <Term
+                                name="contract-type"
```
Comment (fpigeonjr) [inline frontend/src/components/Agreements/AgreementMetaAccordion/AgreementMetaAccordion.jsx:61]: 🤔 Does this need validation?

File: frontend/src/components/ChangeRequests/StatusChangeReviewCard/StatusChangeReviewCard.jsx [hunk 3]
```diff
@@ -30,14 +32,19 @@ function StatusChangeReviewCard({
...
+    forceHover = false,
+    createdById
 }) {
     const keyName = Object.keys(changeTo)[0];
     const totalAmount = useGetBLITotal(bliId);
     const oldCan = useGetNameForCanId(changeTo.can_id?.old);
     const newCan = useGetNameForCanId(changeTo.can_id?.new);
     const { oldValue, newValue } = renderChangeValues(keyName, changeTo, oldCan, newCan);
     const changeMsg = `\u2022 BL ${bliId} ${convertCodeForDisplay("changeToTypes", keyName)}: ${oldValue} to ${newValue}`;
+    const createdBy = useGetUserFullNameFromId(createdById);
```
Comment (fpigeonjr) [inline frontend/src/components/ChangeRequests/StatusChangeReviewCard/StatusChangeReviewCard.jsx:44]: 🤔 should we always handle the budget line creator in the component?

File: frontend/src/components/Agreements/AgreementMetaAccordion/AgreementMetaAccordion.jsx (hunk 2 above)
Comment (fpigeonjr) [reply frontend/src/components/Agreements/AgreementMetaAccordion/AgreementMetaAccordion.jsx:61]: what a minimal Agreement looks like with current validation

image

Comment (fpigeonjr) [reply frontend/src/components/Agreements/AgreementMetaAccordion/AgreementMetaAccordion.jsx:61]: **Service Requirement Type** is a mandatory field when creating an agreement

**Review by johndeange — APPROVED** [review]

File: frontend/src/components/Agreements/AgreementChangesAccordion/AgreementChangesAccordion.jsx (hunk 1 above)
Comment (fpigeonjr) [reply frontend/src/components/Agreements/AgreementChangesAccordion/AgreementChangesAccordion.jsx:11]: yes we will delete the component

File: frontend/src/components/Agreements/AgreementMetaAccordion/AgreementMetaAccordion.jsx (hunk 2 above)
Comment (fpigeonjr) [reply frontend/src/components/Agreements/AgreementMetaAccordion/AgreementMetaAccordion.jsx:61]: will add validation

File: frontend/src/components/ChangeRequests/StatusChangeReviewCard/StatusChangeReviewCard.jsx (hunk 3 above)
Comment (fpigeonjr) [reply frontend/src/components/ChangeRequests/StatusChangeReviewCard/StatusChangeReviewCard.jsx:44]: 💚prob ok for now

**Review by maiyerlee — APPROVED** [review]
Looks good! Thanks for going over part of the PR during last week's Frontend Power Hour (super helpful)!

**Comment by fpigeonjr** [comment]
👀 image

**Pull request merged**